### PR TITLE
Tests the scenario when calling protected static base member from `static do`

### LINKED ISSUE #11929: Calling protected static base member from static do raises MethodAccessException
If the `static do` block of a class attempts to call a `protected static` member of a base class, a MethodAccessException is raised because the code from the do block is moved outside the derived class by the compiler

```fsharp
#nowarn "44" // using Uri.EscapeString just because it's protected static

type C(str : string) =
    inherit System.Uri(str)
    
    static do
        System.Uri.EscapeString("data") |> ignore

C("hello") |> ignore
```
[Sharplab example](https://sharplab.io/#v2:DYLgZgzgNALiBOBXAdlAJiA1AHwMTIHsB3AQ3mQAIAiAFhqooHpGLEIBLZAcwoFV52AOgCiEAMYkADgFMAyjAHcKAKzYwKAI2kS20iuxgByCBUnwCMbZbQUIMEjHZiAsACg3MAJ4yKAYQAUdvAUILYKnFwAlBQAvG4UCfrIABbSAuqynnbSALaC/OyBCpHxiaUJdg5OFGgE5YkJmdl5BSLiUnLh3P5UaA4kVNHYAHz6XITw0m5uAVSpwMAEgxQjYxPSQA===)

**Expected behavior**

The method call should succeed, since the `static do` is the equivalent to C#'s static constructor, or a warning should be produced that while this will compile it won't work.

**Actual behavior**

A `MethodAccessException` is raised on the call

**Known workarounds**
Move the method call into a static member of the derived class, and call it from the `static do` block:
```fsharp
    static do
        C.Do()

    static member internal Do() =
        System.Uri.EscapeString("data") |> ignore
```

### PR DESCRIPTION
**Pull request by edgarfgp**
## Description

It seems that the linked issue not longer happens when `--realsig+` is enabled. So this PR adds tests to demonstrate this

## Checklist

- [x] Test cases added

### CODE REVIEW

**Review by psfinaki — APPROVED** [review]
Thanks for looking into it and testing! 

@KevinRansom @vzarytovskii do we have an ambition to make it work with realsig- as well, or should we just close the issue for good?

File: tests/FSharp.Compiler.ComponentTests/EmittedIL/RealInternalSignature/RealInternalSignature.fs [hunk 1]
```diff
@@ -1253,3 +1253,56 @@ module Test6
...
+        |> asLibrary
+        |> compile
+        |> compileExeAndRun
+        |> shouldSucceed
+        |> withStdOutContainsAllInOrder [
+            "Hello, World"
+        ]
+
+    [<Fact>]
+    let ``Calling protected static base member from `static do` raises MethodAccessException with --realsig-`` () =
```
Comment (KevinRansom) [inline tests/FSharp.Compiler.ComponentTests/EmittedIL/RealInternalSignature/RealInternalSignature.fs:1286]: Why should calling a protected static with realsig- be any different from realsig+  I think it's a bug

**Review by KevinRansom — COMMENTED** [review]
Thank you for highlighting the super-old issue and adding these tests.  I will see what can be done to make realsig- succeed in this case.  We don't really want there to be a behaviour difference between on and off.

**Review by KevinRansom — APPROVED** [review]

File: tests/FSharp.Compiler.ComponentTests/EmittedIL/RealInternalSignature/RealInternalSignature.fs (hunk 1 above)
Comment (edgarfgp) [reply tests/FSharp.Compiler.ComponentTests/EmittedIL/RealInternalSignature/RealInternalSignature.fs:1286]: Thanks Kevin. I did try to make it work but fail miserably so far, so if you could take a look would be amazing.

**Pull request merged**